sql/src/main/java/io/crate/analyze/CreateTableStatementAnalyzer.java

```diff
@@ -71,11 +77,11 @@ public AnalyzedCreateTable analyze(CreateTable<Expression> createTable,
         AnalyzedTableElements<Symbol> analyzedTableElements = TableElementsAnalyzer.analyze(
             analyzedCreateTable.tableElements(), relationName, null);
 
-        // 2nd phase, analyze possible generatedExpressions and defaultExpressions with a reference resolver
+        // 2nd phase, analyze generated/default Expressions and check constraints with a reference resolver
         TableReferenceResolver referenceResolver = analyzedTableElements.referenceResolver(relationName);
         var exprAnalyzerWithReferences = new ExpressionAnalyzer(
             functions, txnCtx, paramTypeHints, referenceResolver, null);
-        List<TableElement<Symbol>> tableElementsWithExpressions = new ArrayList<>(analyzedCreateTable.tableElements().size());
+        List<TableElement<Symbol>> tableElementsWithExpressions = new ArrayList<>((int)(analyzedCreateTable.tableElements().size() * 1.5));
```

Comment:
Why `1.5` as magic multiplier?

Reply:
it is an estimate, a more accurate estimate would be plus (instead of times 1.5) the number of column definitions, because each of those can have a check constraint, so in effect times 2, but I assume most create table statements won't use check constraints.

Reply:
I'd remove the sizing altogether in that case. There are already so many allocations happening here that this kind of micro-optimization won't save the day.

Reply:
removed